ipfs/bitswap/protocol.nim

```diff
@@ -0,0 +1,33 @@
+import pkg/chronos
+import pkg/libp2p/switch
+import pkg/libp2p/stream/connection
+import pkg/libp2p/protocols/protocol
+import ./stream
+
+export stream except readLoop
+
+const Codec = "/ipfs/bitswap/1.2.0"
+
+type
+  BitswapProtocol* = ref object of LPProtocol
+    connections: AsyncQueue[BitswapStream]
+
+proc new*(t: type BitswapProtocol): BitswapProtocol =
```

Comment:
by convention, this is usually done in the `init` method that gets called from the constructor.

Reply:
I went with the naming convention for constructors from [the suggested style guide](https://github.com/status-im/nimbus-eth2/blob/a4941105511eeebfc13fad0281457afb86b79fb5/docs/the_style_book/src/02_formatting.md#naming-conventions).

Reply:
That's not what I meant. The protocol base class defines an init method where the setup of the protocol usually happens, the constructor calls the init method. This is done so that protocols can be inherited from, without having to manually redefine the protocol.